src/Laravel/Eloquent/Filter/DateFilter.php

```diff
@@ -28,13 +31,48 @@ final class DateFilter implements FilterInterface, HasSchemaFilterInterface
      */
     public function apply(Builder $builder, mixed $values, Parameter $parameter, array $context = []): Builder
     {
-        if (!\is_string($values)) {
+        if (!\is_array($values)) {
             return $builder;
         }
 
-        $datetime = new \DateTimeImmutable($values);
+        $operatorValue = [
+            'eq' => '=',
+            'gt' => '>',
+            'lt' => '<',
+            'gte' => '>=',
+            'lte' => '<=',
+        ];
 
-        return $builder->{($context['whereClause'] ?? 'where').'Date'}($this->getQueryProperty($parameter), $datetime);
+        $values = array_intersect_key($values, $operatorValue);
+
+        if (!$values) {
+            return $builder;
+        }
+
+        if (true === ($parameter->getFilterContext()['include_nulls'] ?? false)) {
+            foreach ($values as $key => $value) {
+                $datetime = $this->getDateTime($value);
+                if (null === $datetime) {
+                    continue;
+                }
+                $builder->{$context['whereClause'] ?? 'where'}(function (Builder $query) use ($parameter, $datetime, $operatorValue, $key): void {
+                    $query->whereDate($this->getQueryProperty($parameter), $operatorValue[$key], $datetime)
+                        ->orWhereNull($this->getQueryProperty($parameter));
```

Comment:
You should store the result or `$this->getQueryPtoperty()` in a variable to avoid calling it 2 times (function calls are costly).